packages/react-checkbox/src/components/Checkbox/Checkbox.tsx

```diff
@@ -4,8 +4,8 @@ import { CheckboxBase } from './Checkbox.base';
 import { getStyles } from './Checkbox.styles';
 import { ICheckboxProps, ICheckboxStyleProps, ICheckboxStyles } from './Checkbox.types';
 
-export const Checkbox = styled<
-  ICheckboxProps & React.RefAttributes<HTMLDivElement>,
+export const Checkbox: React.FunctionComponent<ICheckboxProps> = styled<
```

Comment:
I ran into this issue a few times during conversion, but why add explicit typings and generic arguments when they can be inferred? Several styled versions of components I converted to ref-forwarding components lost the `ref` attribute *precisely because* of explicit typings like these.

Reply:
I think the cleanliness improvements to the public API make it worthwhile, and we can help prevent `ref` from getting lost in the future by adding a lint rule.

This also happens to make upgrading to the latest API Extractor version (which we'll eventually need to do) easier because the next minor version beyond where we are right now has decided that inferred types which include `import()` types are "unsupported"...even though it seemed to do fine with them in the past.

Reply:
The reason is to have cleaner/simpler typing:
![image](https://user-images.githubusercontent.com/1207059/93238178-eb3a1100-f735-11ea-9f11-fce72c598cc3.png)

Also, you should not have any problem going forward because now we have `ref` typing added to `IComponentProps` interfaces